integration/projects_list_test.go

```diff
@@ -0,0 +1,143 @@
+package integration
+
+import (
+	"bytes"
+	"fmt"
+	"html/template"
+	"net/http"
+	"net/http/httptest"
+	"net/http/httputil"
+	"os/exec"
+	"strings"
+	"testing"
+
+	"github.com/sclevine/spec"
+	"github.com/stretchr/testify/require"
+)
+
+var _ = suite("projects/list", func(t *testing.T, when spec.G, it spec.S) {
+	var (
+		expect *require.Assertions
+		server *httptest.Server
+	)
+
+	it.Before(func() {
+		expect = require.New(t)
+
+		server = httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+			switch req.URL.Path {
+			case "/v2/projects":
+				auth := req.Header.Get("Authorization")
+				if auth != "Bearer some-magic-token" {
+					w.WriteHeader(http.StatusUnauthorized)
+					return
+				}
+
+				pageNum := req.URL.Query().Get("page")
+				response := projectsListPageOneResponse
+				if pageNum == "2" {
+					response = projectsListPageTwoResponse
+				}
+
+				reqURL := struct {
+					Host string
+				}{
+					Host: req.RemoteAddr,
+				}
+
+				t, err := template.New("response").Parse(response)
+				expect.NoError(err)
+
+				var b []byte
+				buffer := bytes.NewBuffer(b)
+				err = t.Execute(buffer, reqURL)
+				expect.NoError(err)
+
+				w.Write(buffer.Bytes())
+			default:
+				dump, err := httputil.DumpRequest(req, true)
+				if err != nil {
+					t.Fatal("failed to dump request")
+				}
+
+				t.Fatalf("received unknown request: %s", dump)
+			}
+		}))
+	})
+
+	when("all required flags are passed", func() {
+		it.Focus("lists the projects", func() {
+			cmd := exec.Command(builtBinaryPath,
+				"-t", "some-magic-token",
+				"-u", server.URL,
+				"projects",
+				"list",
+			)
+
+			output, err := cmd.CombinedOutput()
+			expect.NoError(err, fmt.Sprintf("received error output: %s", output))
+			expect.Equal(strings.TrimSpace(projectsListOutput), strings.TrimSpace(string(output)))
+		})
+	})
+})
+
+const projectsListOutput = `
+ID                 Owner UUID    Owner ID    Name     Description    Purpose    Environment    Is Default?    Created At              Updated At
+some-project-id    some-uuid     2           magic    hello          none       mars           false          2018-09-27T20:10:35Z    2018-09-27T20:10:35Z
+other-id           some-uuid     2           yes      no             other      venus          true           2018-09-27T20:10:35Z    2018-09-27T20:10:35Z
+`
+const projectsListPageOneResponse = `
+{
+  "projects": [
+    {
+      "id": "some-project-id",
+      "owner_uuid": "some-uuid",
+      "owner_id": 2,
+      "name": "magic",
+      "description": "hello",
+      "purpose": "none",
+      "environment": "mars",
+      "is_default": false,
+      "created_at": "2018-09-27T20:10:35Z",
+      "updated_at": "2018-09-27T20:10:35Z"
+    }
+  ],
+  "links": {
+    "pages": {
+      "first": "https://{{.Host}}/v2/projects?page=1",
+      "next": "https://{{.Host}}/v2/projects?page=2",
+      "last": "https://{{.Host}}/v2/projects?page=2"
+    }
+  },
+  "meta": {
+    "total": 1
```

Comment:
If this is passing, our pagination logic doesn't depend on it, but this should be `2`. It should be the full total, not just what is displayed on the page. E.g:

```
curl -X GET -H "Content-Type: application/json" -H "Authorization: Bearer $DO_TOKEN" "https://api.digitalocean.com/v2/projects?per_page=1" | jq .meta.total
4
```

Despite only showing the details of one project due to `per_page`, the total count is 4.

Reply:
yeah, the meta in the response mostly seems to not matter at all. It focuses mainly on what the `links`say are the next few pages regardless of content size.